xwiki-platform-core/xwiki-platform-uiextension/xwiki-platform-uiextension-api/src/main/java/org/xwiki/uiextension/internal/WikiUIExtension.java

```diff
@@ -78,11 +82,25 @@
 
     private final Provider<XWikiContext> xcontextProvider;
 
+    private final ScriptContextManager scriptContextManager;
+
     /**
      * Parameter manager for this extension.
      */
     private WikiUIExtensionParameters parameters;
 
+    /**
+     * Save the value of "uix" in the {@link XWikiContext} before the initialization, to restore it after the
+     * rendering.
+     */
+    private Map<String, Object> previousUIXContext;
```

Comment:
You cannot have a state in WikiUIExtension since the same instance might be used in different threads, that's why before was returning the value to restore, you will just need to return both values to restore.

Reply:
Hum indeed. I'll use a inner class to return several values.